### DISCUSSION TITLE
Should Supabase communicate potentially breaking changes and have a way to lock versioning?

### DISCUSSION THREAD

**Discussion by rphly** [supabase/supabase#4351] [General]
Hi guys, 
As we know, there's been a couple of issues past few days after the new real-time release, this is expected. 
Kudos to the team for working hard to fix them. But that aside, there were some changes that broke other supabase clients downstream such as the Realtime Dart package (see [issue](https://github.com/supabase/realtime-dart/pull/30) related to 'Boolean values parsed as null').

We have an app that is being used by users out in the wild, and as we all know, pushing an client-side app update is not the most ideal. We would have liked to at least have known that 1. a major update is coming that could potentially break things 2. that we can choose to migrate over slowly but eventually. I understand that not all bugs will be caught, but I think it doesn't really make sense that backend changes can break client side and users also don't have a way to revert to the a previous version. 

From the issue, it seems like the values returned to realtime-dart had changed after the prev update. This could have probably been flagged and at least communicated in some way. 

Hoping to hear what you guys think. I like what Supabase is doing, and I can understand it's early and not as stable as desired, but these are some areas I thought could be improved.

**Comment by w3b6x9** [comment]
@rphly we really appreciate you bringing this up!

I fully agree that we need to do a better job of disseminating information regarding new releases and upcoming breaking changes.

When it comes to `supabase-dart`, and putting aside the recent `Boolean values parsed as null` issue, I created a [GitHub issue](https://github.com/supabase/supabase-dart/issues/56) a little more than a month ago outlining the changes that the new Realtime RLS required of clients. The maintainers of the dart client lib quickly made sure it was compatible with both the new Realtime RLS as well as the previous version of Realtime (non-RLS). We tried to give everyone who relied on `supabase-dart` about a month's time to upgrade but we didn't announce it, and being a dev myself, I fully understand that it's hard to track releases and know when it's crucial to upgrade to a new version when it's not clearly spelled out.

We're trying to do a better job of announcing changes (e.g. [Platform updates: 30 Nov 2021](https://github.com/supabase/supabase/releases/tag/0.20.11.30)) but it's difficult to know the best medium or media to share such information given everyone in the community.

We would love to hear from you, and the rest of the community, as to how you would like to receive communication about future changes.
